### ISSUE TITLE
Support queuing messages when contact is offline

### ISSUE THREAD

**Issue by trufae** [Antidote-for-Tox/Antidote#267]
Sometimes you want to send a message to someone who is offline at the moment, the app should queue those messages and retry to send them when the contact gets online again.

**Comment by dvor** [comment]
This is not reliable solution which will lead to bad UX.

The main issue is that on iOS you cannot run app in background while some users might expect that queued messages would be send after they kill Antidote (or put to background).

Good solution is to add offline messages & push notifications to Tox. See https://github.com/Antidote-for-Tox/Antidote/issues/191 for some discussions.

**Comment by dvor** [comment]
Therefore closing this issue as won't fix.

**Comment by trufae** [comment]
I dont want the app to send the msg in background i want the app to allow me to type stuff even when the user if offline and just automatically send it when it goes online (and i have the app opened)

**Comment by dvor** [comment]
Okay, I'll think about implementing it as an optional feature, turned off by default.

**Comment by trufae** [comment]
Sounds good 👍

**Comment by trufae** [comment]
drop the `wontfix` label

**Comment by dvor** [comment]
Yes, thanks!

**Comment by masvil** [comment]
You mean same like qTox (for Windows, at least), right?

**Comment by dvor** [comment]
Yes, same. When Antidote is active and friends come online, queued messages will be automatically send.

**Comment by masvil** [comment]
Will be way to delete queued message before sending?

**Comment by dvor** [comment]
Yes, I'll implement that feature.

**Comment by masvil** [comment]
OT: so I wonder why qTox doesn't allow it.
